ui/lib/pki/addon/components/pki-key-import.js

```diff
@@ -0,0 +1,61 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { task } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import errorMessage from 'vault/utils/error-message';
+import trimRight from 'vault/utils/trim-right';
+
+// TODO: convert to typescript after https://github.com/hashicorp/vault/pull/18387 is merged
+/**
+ * @module PkiKeyImport
+ * PkiKeyImport components are used to import PKI keys.
+ *
+ * @example
+ * ```js
+ * <PkiKeyImport @model={{this.model}} />
+ * ```
+ *
+ * @param {Object} model - pki/key model.
+ * @callback onCancel - Callback triggered when cancel button is clicked.
+ * @callback onSubmit - Callback triggered on submit success.
+ */
+
+export default class PkiKeyImport extends Component {
+  @service store;
+  @service flashMessages;
+
+  @tracked errorBanner;
+  @tracked invalidFormAlert;
+  @tracked modelValidations;
+
+  @task
+  *submitForm(event) {
```

Comment:
Not sure if this necessary now, but I've seen some concurrency tasks with `waitFor` and I think it's for tests in the future!

Reply:
I'd wondered the purpose in adding that! Thank you!